tsdb/db.go

```diff
@@ -925,6 +926,11 @@ func (db *DB) Appender(ctx context.Context) storage.Appender {
 	return dbAppender{db: db, Appender: db.head.Appender(ctx)}
 }
 
+// AppendExemplar appends an exemplar into the head tsdb assuming that a series ref already exists
+func (db *DB) AppendExemplar(ref storage.SeriesRef, l labels.Labels, e exemplar.Exemplar) (storage.SeriesRef, error) {
```

Comment:
This is not a pattern we follow in TSDB and would be a wrong usage of the appender below. The appender needs to append and then commit to assure it is put into TSDB. We should not be assuming the underlying implementation details. Instead of a helper function like this, I suggest getting an appender from the DB, append the exemplar, and commit it. (same comment for agent/db.go)

Reply:
@codesome , i have removed this piece of code as I shouldn't even need it.